modules/core/src/test/java/org/apache/ignite/internal/encryption/EncryptedCacheCreateTest.java

```diff
@@ -0,0 +1,164 @@
+    /** Non-persistent data region name. */
+    private static final String NO_PERSISTENCE_REGION = "no-persistence-region";
+
+    /** {@inheritDoc} */
+    @Override protected void afterTest() throws Exception {
+        stopAllGrids();
+
+        cleanPersistenceDir();
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTest() throws Exception {
+        cleanPersistenceDir();
+
+        IgniteEx igniteEx = startGrid(0);
+
+        startGrid(1);
+
+        igniteEx.cluster().active(true);
+
+        awaitPartitionMapExchange();
+    }
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String name) throws Exception {
+        IgniteConfiguration cfg = super.getConfiguration(name);
+
+        DataStorageConfiguration memCfg = cfg.getDataStorageConfiguration();
+
+        memCfg.setDataRegionConfigurations(new DataRegionConfiguration()
+            .setMaxSize(10L * 1024 * 1024)
+            .setName(NO_PERSISTENCE_REGION)
+            .setPersistenceEnabled(false));
+
+        return cfg;
+    }
+
+    /** @throws Exception If failed. */
+    public void testCreateEncryptedCache() throws Exception {
+        CacheConfiguration<Long, String> ccfg = new CacheConfiguration<>(ENCRYPTED_CACHE);
+
+        ccfg.setEncryptionEnabled(true);
+
+        IgniteEx grid = grid(0);
+
+        grid.createCache(ccfg);
+
+        IgniteInternalCache<Object, Object> enc = grid.cachex(ENCRYPTED_CACHE);
+
+        assertNotNull(enc);
+
+        KeystoreEncryptionKey key =
+            (KeystoreEncryptionKey)grid.context().encryption().groupKey(CU.cacheGroupId(ENCRYPTED_CACHE, null));
+
+        assertNotNull(key);
+        assertNotNull(key.key());
+    }
+
+    /** @throws Exception If failed. */
+    public void testCreateEncryptedNotPersistedCacheFail() throws Exception {
+        fail("https://issues.apache.org/jira/browse/IGNITE-8640");
+
+        GridTestUtils.assertThrowsWithCause(() -> {
+            CacheConfiguration<Long, String> ccfg = new CacheConfiguration<>(NO_PERSISTENCE_REGION);
+
+            ccfg.setEncryptionEnabled(true);
+            ccfg.setDataRegionName(NO_PERSISTENCE_REGION);
+
+            grid(0).createCache(ccfg);
+        }, IgniteCheckedException.class);
+    }
+
+    /** @throws Exception If failed. */
+    public void testPersistedContentEncrypted() throws Exception {
+        IgniteCache<Integer, String> enc = grid(0).createCache(
+            new CacheConfiguration<Integer, String>(ENCRYPTED_CACHE)
+                .setEncryptionEnabled(true));
+
+        IgniteCache<Integer, String> plain = grid(0).createCache(new CacheConfiguration<>("plain-cache"));
+
+        assertNotNull(enc);
+
+        String encValPart = "AAAAAAAAAA";
+        String plainValPart = "BBBBBBBBBB";
+
+        StringBuilder longEncVal = new StringBuilder(encValPart.length()*100);
+        StringBuilder longPlainVal = new StringBuilder(plainValPart.length()*100);
+
+        for (int i=0; i<100; i++) {
```

Comment:
I think we need spaces after i, after =, etc.  See https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines#CodingGuidelines-A.ConfigureIntelliJIDEAcodestyle